src/Microsoft.Health.Dicom.Functions.UnitTests/DurableFunctionsTimedCleanupTests.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+//using System;
+//using System.Collections.Generic;
+//using System.Text;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Functions.UnitTests
+{
+    public class DurableFunctionsTimedCleanupTests
+    {
+        private readonly ReindexErrorConfiguration _cleanupConfig;
+        private readonly TimerInfo _timer;
+        private readonly ILogger _logger;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly DateTime _definedNow;
+        private readonly DurableFunctionsTimedCleanup _durableFunctionsCleanup;
+        private readonly IDurableOrchestrationClient _durableOrchestrationClientMock;
+
+        public DurableFunctionsTimedCleanupTests()
+        {
+            _cleanupConfig = new ReindexErrorConfiguration();
+            _cleanupConfig.RuntimeStatuses = new OrchestrationRuntimeStatus[] { OrchestrationRuntimeStatus.Completed };
+            _timer = Substitute.For<TimerInfo>(default, default, default);
+            _logger = Substitute.For<ILogger>();
+            _cancellationTokenSource = new CancellationTokenSource();
+            _definedNow = DateTime.UtcNow;
+            _durableFunctionsCleanup = new DurableFunctionsTimedCleanup(
+                Options.Create(_cleanupConfig),
+                () => _definedNow);
+            _durableOrchestrationClientMock = Substitute.For<IDurableOrchestrationClient>();
+        }
+
+        [Fact]
+        public async Task GivenValidInput_CleanupCompletedDurableFunctions__ShouldPurgeListOfOrchestrationsAsync()
+        {
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(givenCondition =>
+                        givenCondition.RuntimeStatus.SequenceEqual(_cleanupConfig.RuntimeStatuses)
+                        && givenCondition.CreatedTimeFrom == DateTime.MinValue
+                        && givenCondition.CreatedTimeTo == _definedNow.AddDays(-_cleanupConfig.MinimumAgeDays)
+                    ),
+                    Arg.Any<CancellationToken>())
+                .Returns(Task.FromResult(new OrchestrationStatusQueryResult { DurableOrchestrationState = Array.Empty<DurableOrchestrationStatus>() }));
+
+            await _durableFunctionsCleanup.Run(_timer, _durableOrchestrationClientMock, _logger, _cancellationTokenSource.Token);
```

Comment:
We also need to check if the proper instance IDs were passed to `IDurableOrchestrationClient.PurgeInstanceHistory` async. We'll need to collect the arguments passed to this method and check the number of calls versus the number of pages.

Eg.
```csharp
var actual = new List<string>();
var expected = new List<string> { "foo", "bar", "baz" };

_durableOrchestrationClientMock
    .When(x => x.PurgeInstanceHistory(Arg.Any<string>())
    .Do(x => actual.Add(x.ArgAt<string>(0));

await _durableFunctionsCleanup.Run(...);

_durableOrchestrationClientMock
    .Received(3)
    .PurgeInstanceHistory(Arg.Is<string>(x => expected.Contains(x));
Assert.Equal(expected, actual);
```